x-pack/platform/plugins/shared/cases/common/types/domain/case/v1.ts

```diff
@@ -95,6 +97,10 @@ const CaseBaseFields = {
    * Observables
    */
   observables: rt.array(CaseObservableRt),
+  /**
+   * Incremental ID
+   */
+  incremental_id: CaseIncrementalId,
```

Comment:
Let's move the `incremental_id` to the `CaseAttributesRt` object. The `CaseBaseFields` are being used by the templates and the configuration type which the `incremental_id` will not be supported.